zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -910,6 +957,36 @@ protected void setLocalSessionFlag(Request si) {
     }
 
     public void submitRequest(Request si) {
+        if (RequestThrottler.enabled()) {
+            enqueueRequest(si);
+        } else {
+            submitRequestNow(si);
+        }
+    }
+
+    public void enqueueRequest(Request si) {
+        if (requestThrottler == null) {
+            synchronized (this) {
+                try {
+                    // Since all requests are passed to the request
+                    // processor it should wait for setting up the request
+                    // processor chain. The state will be updated to RUNNING
+                    // after the setup.
+                    while (state == State.INITIAL) {
+                        wait(1000);
```

Comment:
1 sec feels a little bit ... longer. How did we decide this value? Should it be configurable?

Reply:
this is not invented in this patch :) it has been in the ZooKeeperServer to wait for the request pipeline to start up. We just copy/paste it.